services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/data/HaasController.java

```diff
@@ -0,0 +1,67 @@
+package us.dot.its.jpo.ode.api.controllers.data;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.http.MediaType;
+
+import us.dot.its.jpo.ode.api.accessors.haas.HaasLocationDataRepository;
+import us.dot.its.jpo.ode.api.models.PaginatedGeoJsonResponse;
+import us.dot.its.jpo.ode.api.models.haas.HaasLocation;
+
+@Slf4j
+@RestController
```

Comment:
suggestion(non-blocking): put the base path for the controller at the top of the class to make extending this API simpler (if we need to)
```suggestion
@RestController
@RequestMapping(value = "/data/haas/")
```